DEF[PORTVERSION]=	1.14.0
# ------------------------------------------------------------------------

NAMEBASE=		python-pygit2
VERSION=		${PORTVERSION}
KEYWORDS=		python
VARIANTS=		v11 v12
SDESC[v11]=		Python bindings for libgit2 (3.11)
SDESC[v12]=		Python bindings for libgit2 (3.12)
HOMEPAGE=		https://github.com/libgit2/pygit2
CONTACT=		Python_Automaton[python@ironwolf.systems]

DOWNLOAD_GROUPS=	main
SITES[main]=		PYPI/p/pygit2
DISTFILE[1]=		pygit2-${PORTVERSION}.tar.gz:main

SPKGS[v11]=		single
SPKGS[v12]=		single

OPTIONS_AVAILABLE=	PY311 PY312
OPTIONS_STANDARD=	none
VOPTS[v11]=		PY311=ON PY312=OFF
VOPTS[v12]=		PY311=OFF PY312=ON

# License listed at PyPI
# GPLv2 with linking exception

GENERATED=		yes
DISTNAME=		pygit2-${PORTVERSION}

# install_requires extracted from setup.py
# cffi>=1.16.0
# setuptools; python_version >= "3.12"
# cffi>=1.16.0

[PY311].USES_ON=			python:v11,sutools
[PY311].BUILDRUN_DEPENDS_ON=		python-cffi:single:v11
[PY312].USES_ON=			python:v12,sutools
[PY312].BUILDRUN_DEPENDS_ON=		python-cffi:single:v12

BUILD_DEPENDS=		libgit2:dev:standard
BUILDRUN_DEPENDS=	libgit2:primary:standard
MAKE_ENV=		LIBGIT2={{LOCALBASE}}

post-patch:
	# This version check is always too conservative, causing breakage
	${REINPLACE_CMD} -e '/#error You need a compatible libgit2/d' \
		${WRKSRC}/src/types.h
